src/Parsers/ASTIndexDeclaration.cpp

```diff
@@ -98,4 +99,18 @@ void ASTIndexDeclaration::formatImpl(WriteBuffer & ostr, const FormatSettings &
     }
 }
 
+UInt64 getSecondaryIndexGranularity(const std::shared_ptr<ASTFunction> & type, const ASTPtr & granularity)
+{
+    if (type && type->name == "text")
```

Comment:
Checking `granularity` first makes more sense. That is also what the deleted code in src/Parsers/ParserCreateIndexQuery.cpp did.

```cpp
    if (granularity)
        return granularity->as<ASTLiteral &>().value.safeGet<UInt64>();

    if (type && type->name == "text")
        return ASTIndexDeclaration::DEFAULT_TEXT_INDEX_GRANULARITY;

    if (type && type->name == "vector_similarity")
        return ASTIndexDeclaration::DEFAULT_VECTOR_SIMILARITY_INDEX_GRANULARITY;

    return ASTIndexDeclaration::DEFAULT_INDEX_GRANULARITY;
```

Reply:
New text index doesn't work with arbitrary granularity and ignores it.